python/cuml/feature_extraction/_vectorizers.py

```diff
@@ -261,10 +272,36 @@ def _validate_params(self):
             msg = f"Invalid value for ngram_range={self.ngram_range} {msg}"
             raise ValueError(msg)
 
+        if hasattr(self, "n_features"):
+            if not isinstance(self.n_features, numbers.Integral):
+                raise TypeError(
+                    f"n_features must be integral, got {self.n_features} ({type(self.n_features)})."
+                )
+
     def _warn_for_unused_params(self):
-        if self.analyzer != 'word' and self.stop_words is not None:
-            warnings.warn("The parameter 'stop_words' will not be used"
-                          " since 'analyzer' != 'word'")
+        if self.analyzer != "word" and self.stop_words is not None:
+            warnings.warn(
```

Comment:
Please use the cuml logging functionality as opposed to raw warnings, for example https://github.com/rapidsai/cuml/blob/dfb8877d2708ae3709eae261e3b7f35e969f42c6/python/cuml/tsa/arima.pyx#L664

Reply:
Resolved by https://github.com/rapidsai/cuml/pull/2554/commits/145f59aff7996459080439df7395ac912f2b25ff .